nautobot/extras/models/datasources.py

```diff
@@ -167,3 +174,55 @@ def sync(self, user, dry_run=False):
         if dry_run:
             return enqueue_git_repository_diff_origin_and_local(self, user)
         return enqueue_pull_git_repository_and_refresh_data(self, user)
+
+    @contextmanager
+    def clone(self, path=None, branch=None, head=None, shallow=True):
```

Comment:
Is `clone` too ambiguous of a name? This isn't part of the normal `GitRepository` flow so it should probably have a special name like `clone_to_directory` or something?